test/e2e/tests/test_notebook_instance_lifecycle.py

```diff
@@ -0,0 +1,105 @@
+from e2e import (
+    service_marker,
+    wait_for_status,
+    create_sagemaker_resource,
+    sagemaker_client,
+)
+
+from e2e.bootstrap_resources import get_bootstrap_resources
+import random
+
+from e2e.replacement_values import REPLACEMENT_VALUES
+from time import sleep
+
+
+@pytest.fixture(scope="function")
+def notebook_instance_lifecycleConfig():
+    notebook_instance_lfc_name = random_suffix_name("nblf", 32)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["NOTEBOOK_INSTANCE_LFC_NAME"] = notebook_instance_lfc_name
+    reference, spec, resource = create_sagemaker_resource(
+        resource_plural="notebookinstancelifecycleconfigs",
+        resource_name=notebook_instance_lfc_name,
+        spec_file="notebook_instance_lifecycle_config",
+        replacements=replacements,
+    )
+    assert resource is not None
+    yield (reference, resource, spec)
+    if k8s.get_resource_exists(reference):
+        _, deleted = k8s.delete_custom_resource(reference, 10, 5)
+        assert deleted
+
+
+def get_notebook_instance_lifecycle_config(notebook_instance_lfc_name: str):
+    try:
+        resp = sagemaker_client().describe_notebook_instance_lifecycle_config(
+            NotebookInstanceLifecycleConfigName=notebook_instance_lfc_name
+        )
+        return resp
+    except botocore.exceptions.ClientError as error:
+        logging.error(
+            f"SageMaker could not find a Notebook Instance Lifecycle Configuration with the name {notebook_instance_lfc_name}. Error {error}"
+        )
+        return None
+
+
+@service_marker
+@pytest.mark.canary
+class TestNotebookInstanceLifecycleConfig:
+    def test_CreateUpdateDeleteNotebookLifecycleConfig(
+        self, notebook_instance_lifecycleConfig
+    ):
+        (reference, resource, spec) = notebook_instance_lifecycleConfig
+        assert k8s.get_resource_exists(reference)
+
+        # Getting the resource name
+        notebook_instance_lfc_name = resource["spec"].get(
+            "notebookInstanceLifecycleConfigName", None
+        )
+        assert notebook_instance_lfc_name is not None
+
+        # Verifying that its set correctly
+        lfc_before_update = get_notebook_instance_lifecycle_config(
+            notebook_instance_lfc_name
+        )
+        assert lfc_before_update["OnStart"][0]["Content"] == "cGlwIGluc3RhbGwgUElM"
+        spec["spec"]["onStart"] = [{"content": "cGlwIGluc3RhbGwgc2l4"}]
+        k8s.patch_custom_resource(reference, spec)
+
+        resource = k8s.wait_resource_consumed_by_controller(reference)
+        assert resource is not None
+        sleep(3)  # Done to avoid flakiness
+
+        # Verifying that an update was successful
+        latest_notebook_lf = get_notebook_instance_lifecycle_config(
+            notebook_instance_lfc_name
+        )
+        assert latest_notebook_lf["OnStart"][0]["Content"] == "cGlwIGluc3RhbGwgc2l4"
+
+        # Deleting the resource
+        _, deleted = k8s.delete_custom_resource(reference, 10, 5)
```

Comment:
please keep wait_period to 30 sec for consistency

have a total wait time a little higher than average otherwise, It becomes flaky in canaries otherwise

Reply:
done